### ISSUE TITLE
[BUG] Incorrect learning rate schedule when using multiple GPUs in `MultiModalPredictor`

### ISSUE THREAD

**Issue by SnoopKilla** [autogluon/autogluon#4966]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**

Based on [this line](https://github.com/autogluon/autogluon/blob/e1961672e5d34661ab2e4e78dba64a8741cc88f0/multimodal/src/autogluon/multimodal/optimization/lit_module.py#L384) we have that the maximum number of steps for the training of `MultiModalPredictor` is computed as
```
max_steps = (
    len(self.trainer.datamodule.train_dataloader())
    * self.trainer.max_epochs
    // self.trainer.accumulate_grad_batches
)
```
However, `len(self.trainer.datamodule.train_dataloader())` will return the number of steps in one epoch **before considering the number of GPUs used for training**.

**Example**:
1. I have a dataset of 800 samples and a batch size of 10 -> one epoch will take 80 steps.
1. I train on a machine with 4 GPUs -> effective batch size becomes 40 -> one epoch will take 20 steps.
1. I train for 5 epochs -> `max_steps=100`.
1. `MultiModalPredictor` will compute `max_steps=400` and will set the learning rate schedule accordingly.
1. As a consequence, if I choose `warmup_steps=0.1` (so that 10% of total training steps should be devoted to warmup) and `lr_schedule="cosine_decay"` (so that the learning rate should decay up to a value of 0 at the very last training step), I will actually see that **40% of the training steps are devoted to warmup and after that the learning rate decays slowly up to some positive, non-zero value**.

**Expected behavior**

In the example above, we would expect to see a linear growth in the learning rate for the first 10% of the training and then a cosine decay up to 0 at the very last step.

**To Reproduce**

Minimal example: [gist](https://gist.github.com/SnoopKilla/cba71dd9f2ce99e4371a51b46b872f57)

**Steps to reproduce**:
1. Run the code on a machine with 4 GPUs available.
2. Check the tensorboard logs for the plotted learning rate schedule and assert that it is not as expected.

**Screenshots / Logs**

<img width="951" alt="Image" src="https://github.com/user-attachments/assets/eb2c7cca-752b-4595-aa89-c5cb1e6c05b2" />

**Installed Versions**

<details>

```
INSTALLED VERSIONS
------------------
date                   : 2025-03-07
time                   : 21:52:25.057437
python                 : 3.11.10.final.0
OS                     : Linux
OS-release             : 4.14.355-275.570.amzn2.x86_64
Version                : #1 SMP Sat Nov 30 09:51:35 UTC 2024
machine                : x86_64
processor              : x86_64
num_cores              : 96
cpu_ram_mb             : 382905.51953125
cuda version           : 12.550.127.05
num_gpus               : 4
gpu_ram_mb             : [22500, 22245, 22500, 22500]
avail_disk_size_mb     : 441560

accelerate             : 0.34.2
autogluon              : 1.2
autogluon.common       : 1.2
autogluon.core         : 1.2
autogluon.features     : 1.2
autogluon.multimodal   : 1.2
autogluon.tabular      : 1.2
autogluon.timeseries   : 1.2
boto3                  : 1.35.63
catboost               : 1.2.7
coreforecast           : 0.0.12
defusedxml             : 0.7.1
einops                 : 0.8.0
evaluate               : 0.4.3
fastai                 : 2.7.18
fugue                  : 0.9.1
gluonts                : 0.16.0
huggingface-hub        : 0.26.2
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.4
joblib                 : 1.4.2
jsonschema             : 4.21.1
lightgbm               : 4.5.0
lightning              : 2.4.0
matplotlib             : 3.9.2
mlforecast             : 0.13.4
networkx               : 3.4.2
nlpaug                 : 1.1.11
nltk                   : 3.8.1
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnx                   : None
onnxruntime            : None
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : None
optimum-intel          : None
orjson                 : 3.10.12
pandas                 : 2.2.3
pdf2image              : 1.17.0
Pillow                 : 10.4.0
psutil                 : 6.1.0
pyarrow                : 18.0.0
pytesseract            : 0.3.10
pytorch-lightning      : 2.4.0
pytorch-metric-learning: 2.3.0
ray                    : 2.39.0
requests               : 2.32.3
scikit-image           : 0.24.0
scikit-learn           : 1.5.2
scikit-learn-intelex   : None
scipy                  : 1.14.1
seqeval                : 1.2.2
skl2onnx               : None
spacy                  : 3.7.5
statsforecast          : 1.7.8
tabpfn                 : None
tensorboard            : 2.18.0
text-unidecode         : 1.3
timm                   : 1.0.3
torch                  : 2.5.1+cu124
torchmetrics           : 1.2.1
torchvision            : 0.20.1+cu124
tqdm                   : 4.66.5
transformers           : 4.47.1
utilsforecast          : 0.2.4
vowpalwabbit           : None
xgboost                : 2.1.3
None
```

</details>

**Comment by tonyhoo** [comment]
The issue is fixed and will be released in the next version. thanks for reporting and providing details
